src/vaadin-number-field.html

```diff
@@ -303,13 +302,13 @@
         }
 
         _stepChanged(step) {
-          if (!this.__stepChangedCalled && step === 1) {
-            // Setting the initial value, avoid using it in validation.
-            this.inputElement.step = 'any';
-          } else {
-            this.inputElement.step = step;
-          }
+          // Avoid using initial value in validation
+          const validateByStep = this.__stepChangedCalled
+            || (!this.__stepChangedCalled && this.getAttribute('step') !== null);
```

Comment:
This can be simplified to:
```js
const validateByStep = this.__stepChangedCalled || this.getAttribute('step') !== null;
```

Reply:
If it executes the right side of `||` we already know that `__stepChangedCalled == false`.

Reply:
Thanks! Looks like my brain is on Monday-mode today :P

Reply:
Done.